Server/Source/player_pool.hpp

```diff
@@ -213,7 +213,17 @@ struct PlayerPool final : public IPlayerPool, public NetworkEventHandler, public
 					{
 						return false;
 					}
-					if (!from->areWeaponsAllowed() && 0 < onPlayerGiveTakeDamageRPC.WeaponID && onPlayerGiveTakeDamageRPC.WeaponID <= 47)
+
+					auto slot = WeaponSlotData(onPlayerGiveTakeDamageRPC.WeaponID).slot();
+					if (slot == INVALID_WEAPON_SLOT)
+					{
+						if (onPlayerGiveTakeDamageRPC.WeaponID < 49 || onPlayerGiveTakeDamageRPC.WeaponID > 54)
```

Comment:
What does this check do? It seems to skip some weapon IDs that are only valid for death reasons but then it doesn't skip Connect, Disconnect, Suicide?

Reply:
> What does this check do? It seems to skip some weapon IDs that are only valid for death reasons but then it doesn't skip Connect, Disconnect, Suicide?

It's actually damage reasons and they are not the same with death reasons just because of 200+ IDs (connect, disconnect, suicide).